File: Trabalhos Teóricos/tt02/questao Cometa/Cometa.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>
#include <stdlib.h>

#define NUMENTRADA 1000
#define TAMLINHA   1000

bool isFim(char* s){
    return(strlen(s) >= 1 && s[0] == '0');
}

int proxPassagem(char* s){

    int anoAtual = atoi(s);
    int aux = 76 - ((anoAtual - 1986)%76);
    int anoPassgem;
    if(aux == 0){
        anoPassgem = anoAtual + 76;
    }else{
        anoPassgem = anoAtual + aux;
    }
    return anoPassgem;
}



int main(){

    // inicio leitura pub.in
    char linha[NUMENTRADA][TAMLINHA];
    int numEntrada = 0;

    do
    {
        fgets(linha[numEntrada], TAMLINHA, stdin);
    } while (isFim(linha[numEntrada++]) == false);
    numEntrada--;
    // fim leitura pub.in

    for(int i = 0; i < numEntrada; i++){
        printf("%d\n", proxPassagem(linha[i]));
    }

}

File: Trabalhos Teóricos/tt02/questao Cometa/Cometa.java
public class Cometa {
    public static boolean isFim(String s){
        return(s.length() == 1 && s.charAt(0) == '0');
    }


    public static int proxPassagem(String s){
        int anoAtual = Integer.parseInt(s);
        int aux = 76 - ((anoAtual - 1986)%76);
        int anoPassgem;
        if(aux == 0){
            anoPassgem = anoAtual + 76;
        }else{
            anoPassgem = anoAtual + aux;
        }
        return anoPassgem;
    }

    public static void main(String[] args) {
     
        String[] entrada = new String[1000];
        int numEntrada = 0;

        do{
            entrada[numEntrada] = MyIO.readLine();
        }while(isFim(entrada[numEntrada++]) == false);
        numEntrada--;

        for(int i = 0; i < numEntrada; i++){
            MyIO.println(proxPassagem(entrada[i]));
        }

    }
}
